AmplifyPlugins/Core/AWSPluginsCore/Model/Support/ModelSchema+Extension.swift

```diff
@@ -0,0 +1,24 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+extension ModelSchema {
+
+    public func columnName(forField fieldName: String) -> String {
```

Comment:
the scenarios we talked about for DataStore queries apply for API queries as well
1. If the field cannot be resolved to the ModelField, then use it as is, so `let columnName = fieldName`.This scenario is when the developer has explicitly specified the correct column name to use such as "commentPostId" or "postID".
2. If the field can be resolved to the ModelField, then retrieve it's association, if it is a `belongsTo` or `hasOne` then return targetName
3. If for 2, there is no targetName available, then default to the expected column name provisioned by AppSync
```
let defaultFieldName = modelSchema.name.camelCased() + fieldName.pascalCased() + "Id"
```

Reply:
take a look at `getFieldNameForAssociatedModels` for more info